src/components/draw/DrawDirective.js

```diff
@@ -439,46 +788,49 @@ goog.require('ga_map_service');
               }
 
               if (!scope.adminShortenUrl) {
-                $timeout(function() {
-                  updateShortenUrl(layer.adminId);
-                }, 0, false);
-               }
+                updateShortenUrl(layer.adminId);
+              }
             });
           };
           var saveDebounced = gaDebounce.debounce(save, 133, false, false);
 
+          $rootScope.$on('$translateChangeEnd', function() {
+            if (layer) {
+              layer.label = $translate.instant('draw');
+            }
+          });
+
 
-          // Utils
-          // Find the first feature from a vector layer
-          var findDrawnFeature = function(pixel, vectorLayer) {
-              var featureFound;
-              map.forEachFeatureAtPixel(pixel, function(feature, olLayer) {
-                featureFound = feature;
-              }, this, function(olLayer) {
-                return (layer == olLayer);
-              });
-              return featureFound;
-            };
 
+          ////////////////////////////////////
+          // Utils functions
+          ////////////////////////////////////
           // Change cursor style on mouse move, only on desktop
           var updateCursorStyle = function(evt) {
-            var feature = findDrawnFeature(map.getEventPixel(evt));
-            map.getTarget().style.cursor = (feature) ? 'pointer' : '';
+            var featureFound = map.forEachFeatureAtPixel(evt.pixel,
+                function(feature, olLayer) {
+              return feature;
+            }, this, function(olLayer) {
+              return (layer == olLayer);
+            });
+            var isSketchFeature = !!featureFound && !featureFound.getStyle();
+            map.getTarget().style.cursor = (featureFound) ?
+                ((isSketchFeature) ? 'move' : 'pointer') : '';
           };
           var updateCursorStyleDebounced = gaDebounce.debounce(
               updateCursorStyle, 10, false, false);
 
-          // Focus on the first input.
-          var setFocus = function() {
-            $timeout(function() {
-              var inputs = element.find('input, select');
-              if (inputs.length > 0) {
-                inputs[0].focus();
-              }
-            });
-          };
+          // Fix position of dropdown
+          element.find('.dropdown-toggle').click(function() {
+            var bt = $(this);
+            var dropdown = bt.next('.dropdown-menu');
+            var dropDownTop = bt.offset().top + bt.outerHeight() -
+                50; // 50 seems to be the size of the #header
+            dropdown.css('top', dropDownTop + 'px');
+            dropdown.css('left', bt.offset().left -
+                (dropdown.outerWidth() - bt.outerWidth()) + 'px');
+          });
```

Comment:
It's a huge link function. Maybe move definition up into general definitions and we would gain 2 levels of indents. And probably further splitting possible. We'll see.